# EurasianOtter_PopGen

Bioinformatic scripts for population genomic analyses of Eurasian otters from Britain. 
First author email <duplessiss@cardiff.ac.uk>.

### Citation:
Sarah J du Plessis, Mark Blaxter, Klaus-Peter Koepfli, Elizabeth A Chadwick, Frank Hailer, Genomics reveals complex population history and unexpected diversity of Eurasian otters (*Lutra lutra*) in Britain relative to genetic methods, *Molecular Biology and Evolution*, 2023;, msad207, <https://doi.org/10.1093/molbev/msad207>

### Abstract:
Conservation genetic analyses of many endangered species have been based on genotyping of microsatellite loci and sequencing of short fragments of mtDNA. The increase in power and resolution afforded by whole genome approaches may challenge conclusions made on limited numbers of loci and maternally inherited haploid markers. Here we provide a matched comparison of whole genome sequencing versus microsatellite and control region genotyping for Eurasian otters (Lutra lutra). Previous work identified four genetically differentiated ‘stronghold’ populations of otter in Britain, derived from regional populations that survived the population crash of the 1950-80s. Using whole genome resequencing data from 45 samples from across the British stronghold populations we confirmed some aspects of population structure derived from previous marker-driven studies. Importantly we showed that genomic signals of the population crash bottlenecks matched evidence from otter population surveys. Unexpectedly, two strongly divergent mitochondrial lineages were identified that were undetectable using control region fragments, and otters in the east of England were genetically distinct and surprisingly variable. We hypothesise that this previously unsuspected variability may derive from past releases of Eurasian otters from other, non-British source populations in England around the time of the population bottleneck. Our work highlights that even reasonably well studied species may harbour genetic surprises, if studied using modern high-throughput sequencing methods.

### Raw data:
Raw sequence reads are available from the Darwin Tree of Life Data Portal (https://portal.darwintreeoflife.org/data/root/details/Lutra%20lutra).

All whole mitochondrial genome sequences included in this study are found in 'Whole mitochondrial genome_All sequences_n61.fasta'. 

